controller/hybridgateway/errors/errors.go

```diff
@@ -0,0 +1,19 @@
+package errors
+
+import (
+	"fmt"
+)
+
+var (
+	// ErrNoGatewayFound is returned when a referenced Gateway does not exist in the cluster.
+	ErrNoGatewayFound = fmt.Errorf("no supported gateway found")
+
+	// ErrNoGatewayClassFound is returned when a GatewayClass referenced by a Gateway does not exist in the cluster.
+	ErrNoGatewayClassFound = fmt.Errorf("no gatewayClass found for gateway")
+
+	// ErrNoGatewayController is returned when a GatewayClass exists but is not controlled by this controller.
```

Comment:
nit: would use managed instead of controlled, still a bit opinionated, feel free to just dismiss (I know, it was even just moved code) :-)

```suggestion
	// ErrNoGatewayController is returned when a GatewayClass exists but is not managed by this controller.
```